### ISSUE TITLE
Incorrect skiplink error

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#1346]
On the following page, axe-core incorrectly flagged a link with `href="#0"` as a skiplink with no target. The link is something with an event handler on it, the href isn't supposed to reference something. Not sure how to work around that though, except to put it in needs review.
https://css-tricks.com/reacts-experimental-suspense-api-will-rock-for-fallback-ui-during-data-fetches/

**Comment by scurker** [comment]
Are you suggesting that we put all links with an `href="#..."` value that does not point to a target as needs review?

**Comment by straker** [comment]
What if we only looked at the first link on the page and made sure the href started with `#...` and that it pointed to a target that was visible. WebAIM states that

> The horizontal location of the link doesn't matter much. The link can be on the left side, the middle, the right, or in some location in between. The key is to make sure the link is one of the first items that screen readers hear and that keyboard users tab to. Otherwise, users may not realize there is a "skip navigation" link there at all, and may waste time trying to muddle through the extraneous links. Screen reader users especially may get impatient if they don't hear the "skip navigation" link very early in the page.

**Comment by straker** [comment]
As an alternative solution if we want to look at all skip links on a page, we could define a skip link as any anchor element whose href starts with `#...` and which precedes the first link whose href doesn't start with `#...`

```js
let firstPageLink = document.querySelector('a:not([href^="#"])');
let skipLinks = Array.from(document.querySelectorAll('a[href^="#"]')).filter(link => {
  return link.compareDocumentPosition(firstPageLink) === 4
});
```

**Comment by WilcoFiers** [comment]
Cool idea! Yeah, let's do it!
